cpp/src/arrow/flight/meson.build

```diff
@@ -64,6 +64,7 @@ flight_proto_files = custom_target(
     ],
 )
 
+grpc_cpp = dependency('grpc++')
```

Comment:
It seems that this isn't used in this PR.

Reply:
Surprisingly it is, because using this as a dependency will ensure that override_find_program is called for the grpc_cpp_plugin. Without this, if you don't have that installed on your system then the build will fail

Reply:
Wow!